definitions/npm/styled-components_v4.x.x/flow_v0.104.x-/styled-components_v4.x.x.js

```diff
@@ -182,7 +182,7 @@ declare module 'styled-components' {
   |}
 
   declare export type StyledShorthandFactory<V> = {|
-    [[call]]: <StyleProps, Theme>(string[], ...Interpolation<PropsWithTheme<StyleProps, Theme>>[]) => StyledComponent<StyleProps, Theme, V>;
+    [[call]]: <StyleProps, Theme>(string[] | func, ...Interpolation<PropsWithTheme<StyleProps, Theme>>[]) => StyledComponent<StyleProps, Theme, V>;
```

Comment:
What is `func`, and where does it come from? Flow should error here, if it's not erroring it might be a bug in Flow.

Reply:
Also, I would suggest separating/overloading the call signatures, since that is probably more in line with how the API behaves:

```suggestion
    [[call]]: <StyleProps, Theme>(string[], ...Interpolation<PropsWithTheme<StyleProps, Theme>>[]) => StyledComponent<StyleProps, Theme, V>;
    [[call]]: <StyleProps, Theme>((props: PropsWithTheme<StyleProps, Theme>) => Interpolation<any>) => StyledComponent<StyleProps, Theme, V>;
```

:warning: **NOTE** I am not sure if the return in `(props: PropsWithTheme<StyleProps, Theme>) => Interpolation<any>` makes sense? What should be returned from this function? A JSS-like object, or something else?

Reply:
Okay, this change worked great in my local instance! Thank you!
Going to commit this and see how it does with the tests.